### PR TITLE
CLOUDP-100528: Allow custom context for the store

### PR DESCRIPTION
**Pull request by fmenezes**
## Proposed changes

Allow custom context for the store, the idea here is to cascade context from main all the way through the commands, this is part 1 receiving context on the store, subsequent PRs will add the passing of the context from the commands into the store.

_Jira ticket:_ CLOUDP-100528

## Checklist

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added any necessary documentation (if appropriate)
- [x] I have updated [e2e/E2E-TESTS.md](e2e/E2E-TESTS.md) (if a new command or e2e test has been added)
- [x] I have run `make fmt` and formatted my code

### CODE REVIEW

File: internal/store/store.go [hunk 1]
```diff
@@ -206,6 +208,14 @@ func WithAuthentication(c CredentialsGetter) Option {
 	}
 }
 
+// WithContext sets the store context.
+func WithContext(ctx context.Context) Option {
```
Comment (fmenezes) [inline internal/store/store.go:212]: This is the main change

**Review by gssbzn — DISMISSED** [review]
LGTM, thanks for all the work put into this and the low impact approach 💯

**Review by gssbzn — APPROVED** [review]
LGTM

**Review by andreaangiolillo — APPROVED** [review]
LGTM

**Pull request merged**